counsel.el

```diff
@@ -5186,16 +5186,37 @@ N in your system."
   "List of potential build subdirectory names to check for."
   :type '(repeat directory))
 
+(defvar counsel-compile-phony-pattern
+  (rx (: bol ".PHONY:" space (group (one-or-more print)) eol))
+  "Regexp for extracting PHONY targets from Makefiles.")
+
 ;; This is loosely based on the Bash Make completion code
+(defun counsel-compile--probe-make-targets (dir)
+  "Return a list of Make targets for DIR.
+
+If the make errors we return an empty list.  This is usually because
+some sort of configuration needs to be done first or the source tree
+is pristine and being used for multiple build trees."
+  (let ((default-directory dir)
+        (targets))
+    (with-temp-buffer
+      ;; 0 = no-rebuild, -q & 1 needs rebuild, 2 error
+      (when (> 2 (call-process "make" nil t nil "-nqp"))
```

Comment:
Is this condition cross-platform? Or is specific to, say, GNU Make?

Reply:
Good point. According to https://www.freebsd.org/cgi/man.cgi?make(1) the 0/1 distinction is the same but I couldn't see what an outright error would report. I'll see if I can find a BSD system to play with...